Components v2!

naem-less opened the pull request:
## What does this PR do?
Changes most messages to use components v2. Some messages don't need to be cv2 since they aren't shown to the general user and if they are they wont really benefit from cv2 unless you just add a container and call it a day. However if consistency is necessary it's not that big of a job to make them use cv2

What still needs to be done is making the new `ui.LayoutView`s persistent as were their old `ui.View` counterparts. Although due to the fact that the messages these views are attached to contained dynamic elements its not that simple (unless I'm overlooking something obvious). My idea is to use `ui.DynamicItem` for them but I'm not very well versed in them yet, so ill do testing on a personal bot of mine and if i figure something out I'll commit it to this PR. Of course if someone can think of a different way they can suggest it.
## Fill out the following

- [x] If code changes were made then they have been tested.
- [ ] This PR fixes a bug or issue.
- [ ] This PR adds a new feature.
- [ ] This PR is not a code change (i.e, README)

naem-less committed:
Use components v2

naem-less committed:
Use view timeouts to edit the message the components are on and disable said components, instead of creating persistent views
The timeout durations are decently generous, and in the real server the buttons likely wont prematurely be disabled before the user had time to interact with them

naem-less committed:
Add handling on the chance that if the `.from_message()` method wasn't called, editing the message wont give an error

Sacul0457 commented on cogs/readthedamnrules.py:
```diff
@@ -29,15 +30,22 @@ async def get_messages_to_move(self, reference_message: discord.Message) -> list
+        # cv2 for rtdr is tempting, but because the messages get deleted and a media gallery needs a valid url to display images
+        # the images will eventually show up as invalid
+        # unless we do some sapphire images tomfoolery we'll have to compromise
+        gallery_items: ui.MediaGallery = []
```
```suggestion
        gallery_items: ui.MediaGallery = []
        files: list[discord.File] = []
```
see below

Sacul0457 commented on cogs/readthedamnrules.py:
```diff
@@ -29,15 +30,22 @@ async def get_messages_to_move(self, reference_message: discord.Message) -> list
+                    description=attachment.description
+                )
+                gallery_items.append(item)
+        return gallery_items
```
```suggestion
                file = await attachment.to_file()
                files.append(file)
                item = discord.MediaGalleryItem(
                    file.uri,
                    description=attachment.description
                )
                gallery_items.append(item)
        return gallery_items, files
```
actually, you can use a file here, see [here](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=mediagalleryitem#discord.MediaGalleryItem). You basically need to send the files with the view, so make sure to edit the rest

Sacul0457 reviewed (changes requested):
> What still needs to be done is making the new ui.LayoutViews persistent as were their old ui.View counterparts. Although due to the fact that the messages these views are attached to contained dynamic elements its not that simple (unless I'm overlooking something obvious). My idea is to use ui.DynamicItem for them but I'm not very well versed in them yet, so ill do testing on a personal bot of mine and if i figure something out I'll commit it to this PR. Of course if someone can think of a different way they can suggest it.

Cv2 views work the same as normal views, they didn't change. Currently in the PR, I don't see how persistency is an issue..., could you point them out for me pls :)

Also kiki has this [cv2 branch](https://github.com/kiki0124/Sapphire-helper/tree/CV2)

naem-less committed:
Make all views persistent across bot restarts again

naem-less commented:
> > What still needs to be done is making the new ui.LayoutViews persistent as were their old ui.View counterparts. Although due to the fact that the messages these views are attached to contained dynamic elements its not that simple (unless I'm overlooking something obvious). My idea is to use ui.DynamicItem for them but I'm not very well versed in them yet, so ill do testing on a personal bot of mine and if i figure something out I'll commit it to this PR. Of course if someone can think of a different way they can suggest it.
> 
> Cv2 views work the same as normal views, they didn't change. Currently in the PR, I don't see how persistency is an issue..., could you point them out for me pls :)
> 
> Also kiki has this [cv2 branch](https://github.com/kiki0124/Sapphire-helper/tree/CV2)

I see what you mean. I cant event remember how i managed to overlook this, but ill attribute it to being tired since i did make the final touches at midnightly hours

naem-less committed:
Improve `get_media_gallery_items` to also return a list of `discord.File`'s to send with the initial post message

naem-less committed:
Improve logic related to mentioning the OP of a support post
Depending on whether the command executor is expert+ for the `/unrelated` and `/incomplete-post` commands the OP will either be mentioned or not be mentioned

naem-less committed:
Handle components v2 quick reply deletions

Sacul0457 commented on cogs/epi.py:
```diff
@@ -19,20 +19,57 @@
-    def __init__(self):
-        super().__init__(timeout=None)
-    
+class GetNotified(ui.ActionRow):
```
```suggestion
class GetNotifiedButton(ui.ActionRow):
```

naem-less committed:
some performance improvements

naem-less committed:
Handle possible exceptions when fetching the epi message in each post

Sacul0457 reviewed (changes requested):
For views with `timeout=None`, we no longer need to attach a message to the view and handle `on_timeout` since it will never timeout.

naem-less commented:
> For views with `timeout=None`, we no longer need to attach a message to the view and handle `on_timeout` since it will never timeout.

True, another amazing oversight on my part. Im nowhere near home so ill only fix this by Wednesday or later
